firebase-storage/src/main/java/com/google/firebase/storage/FirebaseStorage.java

```diff
@@ -307,4 +322,9 @@ public FirebaseApp getApp() {
   InternalAuthProvider getAuthProvider() {
     return mAuthProvider != null ? mAuthProvider.get() : null;
   }
+
+  @Nullable
+  EmulatedServiceSettings getEmulatorSettings() {
```

Comment:
Would it make sense to "generify" this support and just expose host, port and URL? This will make it less likely that I will abuse this feature for non-Emulator settings in a future PR.

Reply:
Maybe? But that means I have to pass two things all the way down to the request instead of having one thing to encapsulate it.

Reply:
You can pass a Uri. This could then return the host. You might even be able to replace all of `EmulatorSettings` with a `Uri host`.